### ISSUE TITLE
The Copy() method of Metric does not copy the metric type nor the aggregate boolean

### ISSUE THREAD

**Issue by mirath** [influxdata/telegraf#3562]
## Bug report
Using the Copy() method of the Metric interface does not copy the metric type (the value returned by the Type() method) nor the aggregate boolean (the value returned by the IsAggregate() method).

This is counterintuitive, a deep copy should produce an identical copy. Is there a reason these values are not considered by the Copy() method?

### System info:
Branch based of commit 206397d4755e005 from Oct 24 2017.

### Steps to reproduce:
This Go program can reproduce the issue

```go
package main

import (
	"time"
	"fmt"
	"reflect"
	"github.com/influxdata/telegraf/metric"
)

func main() {
	var m, _ = metric.New(
		"test",
		map[string]string{"tag_name": "tag_value1"},
		map[string]interface{}{"a": float64(1)},
		time.Now(),
	)
	m.SetAggregate(true)

	copy := m.Copy()
	fmt.Println("metric:", m, "copy:", copy)
	fmt.Println("The reflec library DeepEqual function returns false:", reflect.DeepEqual(m, copy))
	fmt.Println("metric Type:", m.Type(), "; copy Type:", copy.Type())
	fmt.Println("metric IsAggregate:", m.IsAggregate(), "; copy IsAggregate:", copy.IsAggregate())
}
```

### Expected behavior:
Calls for the IsAggregate and Type functions return the same value for both the original metric and its deep copy. The DeepEqual method from the reflect library should always return true when comparing a metric with its deep copy

### Actual behavior:
Calls for the IsAggregate and Type functions return different values for both the original metric and its deep copy. The DeepEqual method from the reflect library may return false when comparing a metric with its deep copy

**Comment by mirath** [comment]
If this behaviour is actually a bug, can I give it a crack at fixing it? I am using the Copy() method as part of the test in a feature I'm working on (TopK processor), and I'd like to get this fix asap

**Comment by danielnelson** [comment]
If you can fix it that would be great, thank you

**Comment by mirath** [comment]
I've added the pertinent changes to the PR 3400. Let me know if the fix is ok with the proyect's guidelines and direction

**Comment by danielnelson** [comment]
Yeah looks good.

**Comment by danielnelson** [comment]
Closing, this was fixed in 1.6
